amplfi/train/callbacks/testing.py

```diff
@@ -212,13 +230,42 @@ def on_test_batch_end(
             outpath=skymap_filename,
         )
 
+        if reweighted is not None:
+            reweighted_outdir = outdir / "reweighted"
+            skymap_filename = reweighted_outdir / "mollview.png"
+            reweighted_outdir.mkdir(exist_ok=True)
+            reweighted.plot_mollview(
+                self.nside,
+                outpath=skymap_filename,
+            )
+
     def on_predict_batch_end(
         self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx=0
     ):
-        return self.on_test_batch_end(
-            trainer, pl_module, outputs, batch, batch_idx, dataloader_idx
+        # test_step returns bilby result object
+        # and optionally a reweighted result
+        result: "AmplfiResult"
+        reweighted: Optional["AmplfiResult"]
+        result, reweighted = outputs
+
+        gpstime = batch[2].cpu().numpy()[0]
```

Comment:
Is there a nicer way to avoid the index `[2]`? If not, then add a FIXME comment here.